membership-attribute-service/app/controllers/AttributeController.scala

```diff
@@ -129,17 +128,34 @@ class AttributeController extends Controller with LoggingWithLogstashFields {
           subs
         }
       }
+
     }
 
-    val attributes: DisjunctionT[Future, String, Option[Attributes]] = for {
-      accounts <- EitherT(withTimer(s"ZuoraAccountIdsFromIdentityId", zuoraRestService.getAccounts(identityId)))
+    def zuoraAccountsQuery(identityId: String): Future[Disjunction[String, QueryResponse]] = zuoraRestService.getAccounts(identityId).map {
+      _.leftMap { error =>
+        log.error(s"Calling ZuoraAccountIdsFromIdentityId failed for identityId ${identityId}. Error: ${error}")
+        error
+      }
+    }
+
+    val attributes = for {
+      accounts <- EitherT[Future, String, QueryResponse](withTimer(s"ZuoraAccountIdsFromIdentityId", zuoraAccountsQuery(identityId)))
       accountIds = queryToAccountIds(accounts)
-      subscriptions <- EitherT[Future, String, List[Subscription[AnyPlan]]](withTimer(s"ZuoraGetSubscriptions", getSubscriptions(accountIds)))
+      subscriptions <- EitherT[Future, String, List[Subscription[AnyPlan]]](
+        if(accountIds.nonEmpty) withTimer(s"ZuoraGetSubscriptions", getSubscriptions(accountIds))
+        else Future.successful(\/.left(s"No account ids for $identityId and so no subscriptions either"))
+      )
+
     } yield {
       AttributesMaker.attributes(identityId, subscriptions, LocalDate.now())
     }
 
-    attributes.run.map(_.toOption).map(_.getOrElse(None))
+    attributes.run.map {
+      _.leftMap { errorMsg =>
+        log.info(s"Tried to get Attributes for ${identityId} but failed with $errorMsg")
```

Comment:
same for here, I'd use the pattern above to add to the error message and pass it on to the caller to decide what to do